source/simulator.hpp

```diff
@@ -0,0 +1,23 @@
+#pragma once
+
+#include <string>
+#include <unordered_map>
+
+#include "scheduler.hpp"
+
+namespace sim {
+
+class Simulator {
+public:
+    Simulator();
+    void create_device(std::string a_name);
+    void create_link(std::string a_node1, std::string a_node2, size_t a_delay);
+    void start();
+
+private:
+    Scheduler m_scheduler;
```

Comment:
Usage of `Scheduler` copy is impossible (because `Scheduler` has no copy constructor).
```suggestion
    Scheduler& m_scheduler;
```